scripts/ci/check_compliance.py

```diff
@@ -1447,6 +1447,64 @@ def run(self):
             with open(file, "r") as fp:
                 self.check_file(file, fp)
 
+class FileEncoding(ComplianceTest):
+    """
+    Check the encoding for source files.
+    """
+    name = "FileEncoding"
+    doc = "Check the encoding for source files."
+    path_hint = "<git-top>"
+
+    MATCH_FILES = [
+            r'.*\.c$',
+            r'.*\.h$',
+            r'.*\.sh$',
+            r'.*\.yaml$',
+            r'.*Kconfig.*',
+            r'.*CMakeLists.txt$',
+            ]
+
+    def match_file(self, file):
+        for match in self.MATCH_FILES:
+            if re.match(match, file):
+                return True
+
+        return False
+
+    def check_file(self, file, fp):
+        if not self.match_file(str(file)):
+            return
+
+        for line_num, line in enumerate(fp.readlines(), start=1):
+            line = line.strip()
+
+            if not line:
+                continue
+
+            if "Copyright" in line:
+                continue
+
+            try:
+                line.encode().decode(encoding="ascii", errors="strict")
```

Comment:
```suggestion
                line.encode(encoding="latin-1", errors="strict")
```

OK  -> '// Copyright (c) 2023 Benjamin Cabé <benjamin@zephyrproject.org>'
OK -> '// Copyright (c) 2023, Benjamin Björnsson <benjamin.bjornsson@gmail.com>'
Fail -> '// Copyright (c) 2023, ❤❤❤'

With latin-1 micro & degree etc are accepted, and most of the remaining errors are weird quotes etc. where comments have been copied from datasheets.

Reply:
FWIW I believe the way @fabiobaltieri implemented the check, a line containing "Copyright" is skipped altogether (L1484-1485), so can contain anything really.
But that being said, I agree that we need to open it up to a bit more than just 7-bit ASCII, for the reason you're indicating among others.

Reply:
Yeah, I'd rather dodge argument about how to spell your name or your company name. May do latin-1 for these lines but then that's also somewhat restrictive for other countries and gets dangerously into inclusiveness territory.

Reply:
> where comments have been copied from datasheets

Oh that explains, good point.